idea-plugin/src/main/java/com/palantir/javaformat/intellij/PalantirJavaFormatFormattingService.java

```diff
@@ -101,18 +102,21 @@ public void run() {
                             request.getDocumentText().length(),
                             request.getFormattingRanges()));
                 }
-                List<Replacement> replacements =
-                        formatterService.get().getFormatReplacements(request.getDocumentText(), toRanges(request));
-                if (logger.isDebugEnabled()) {
-                    logger.debug(String.format(
-                            "Applying %s replacements with ranges=%s",
-                            replacements.size(),
-                            replacements.stream()
-                                    .map(Replacement::getReplaceRange)
-                                    .collect(Collectors.toSet())));
+
+                // Note: I attempted to implement this using getFormatReplacements and elide an update when
+                // there were no replacements. There is a bug in the underlying formatter where it _always_
+                // returns a change. Thus we must perform a content aware diff / branching.
+                String formattedText = applyReplacements(
+                        request.getDocumentText(),
+                        formatterService.get().getFormatReplacements(request.getDocumentText(), toRanges(request)));
```

Comment:
We should probably use the `preFormatText` instead:

```suggestion
                String formattedText = applyReplacements(
                        preFormatText,
                        formatterService.get().getFormatReplacements(preFormatText, toRanges(request)));
```


same in the in debug message above: https://github.com/palantir/palantir-java-format/pull/1188/files?new_files_changed=true#diff-65737b320acf0cb984bb55ee91a1aee8fd41a4b53051c53f277e652effb9f85bR102